components/org.wso2.carbon.identity.user.ekyc/src/main/java/org/wso2/carbon/identity/user/ekyc/UserEKYCConnectorImpl.java

```
@@ -0,0 +1,182 @@
+    @Override
+    public EKYCSessionDTO generateNewSession(String userId, int tenantId, String service, List<String> claims) throws
+            UserEKYCException, IDVException, ConfigurationManagementException {
+        EKYCSessionDTO ekycSessionDTO = getIdvService().generateNewSession(service, claims);
+        EKYCVerifiedCredentialDAO.getInstance().createUserEKYC(
+                ekycSessionDTO.getSessionId(),
+                userId,
+                tenantId,
+                UserEKYCConstants.EKYCProccessStatuses.PENDING,
+                null);
+        log.debug("Generated session: " + ekycSessionDTO.toString());
+        return ekycSessionDTO;
+    }
+
+    @Override
+    public List<EKYCVerifiedCredentialDTO> getVerifiedCredentials(String userId, int tenantId) throws
+            UserEKYCException {
+        List<EKYCVerifiedCredentialDTO> ekycVerifiedCredentialDTOs = EKYCVerifiedCredentialDAO
+                .getInstance().getUserEKYCVCs(userId, tenantId);
+        log.debug("User VCs " + Arrays.toString(ekycVerifiedCredentialDTOs.toArray()));
+        return ekycVerifiedCredentialDTOs;
+    }
+
+    @Override
+    public void deleteVerifiedCredential(String sessionId, String userId, int tenantId) throws UserEKYCException {
+        EKYCVerifiedCredentialDAO
+                .getInstance().deleteUserEKYCVC(sessionId, userId, tenantId);
+        log.debug("Deleted VC: " + sessionId);
+    }
+
+    @Override
+    public EKYCVerifiedCredentialDTO getPendingVerifiedCredential(String sessionId, String userId, int tenantId)
+            throws UserEKYCException, IDVException, ConfigurationManagementException {
+        EKYCVerifiedCredentialDTO ekycVerifiedCredentialDTO = EKYCVerifiedCredentialDAO
+                .getInstance().getUserEKYCVC(sessionId, userId, tenantId);
+        if (ekycVerifiedCredentialDTO != null && isVerifiedCredentialPending(ekycVerifiedCredentialDTO)) {
+            JsonObject vc = getIdvService().getSessionVc(userId, sessionId);
```

If the http client creation fails, this should throw an error, right?

---

I added exception throw during http client generation so you can't create instance without properly initialized http client